robot-core/src/main/java/org/obolibrary/robot/ExplainOperation.java

```diff
@@ -56,6 +58,117 @@
+   * @param maxUnsat cutoff - limit number of tested unsatisfiable classes to maxUnsat classes
+   * @return a set of explanations
+   */
+  public static Set<Explanation<OWLAxiom>> explainUnsatisfiableClasses(
+      OWLOntology ontology,
+      OWLReasoner reasoner,
+      OWLReasonerFactory reasonerFactory,
+      int max,
+      int maxUnsat) {
+    List<OWLClass> unsatisfiable_classes =
+        new ArrayList<>(reasoner.getUnsatisfiableClasses().getEntitiesMinusBottom());
+    Collections.sort(unsatisfiable_classes);
+    if (maxUnsat > 0 && unsatisfiable_classes.size() > maxUnsat) {
+      unsatisfiable_classes = unsatisfiable_classes.subList(0, maxUnsat);
+    }
+    return getUnsatExplanationsForClasses(ontology, reasonerFactory, max, unsatisfiable_classes);
+  }
+
+  /**
+   * Compute explanations for all root unsatisfiable classes See
+   * https://github.com/matthewhorridge/owlexplanation/blob/439c5ca67835f5e421adde725e4e8a3bcd760ac8/src/main/java/org/semanticweb/owl/explanation/impl/rootderived/StructuralRootDerivedReasoner.java#L122
+   *
+   * @param ontology the ontology to be tested
+   * @param reasoner the reasoner to be used to determine the unsatisfiable classes
+   * @param reasonerFactory the reasoner factory to be used to compute the explanations
+   * @param max maximum number of explanations to be computed
+   * @return a set of explanations
+   */
+  public static Set<Explanation<OWLAxiom>> explainRootUnsatisfiableClasses(
+      OWLOntology ontology, OWLReasoner reasoner, OWLReasonerFactory reasonerFactory, int max) {
+    RootDerivedReasoner rootReasoner =
+        new StructuralRootDerivedReasoner(
+            ontology.getOWLOntologyManager(), reasoner, reasonerFactory);
+    List<OWLClass> unsatisfiable_classes =
+        new ArrayList<>(rootReasoner.getRootUnsatisfiableClasses());
+    return getUnsatExplanationsForClasses(ontology, reasonerFactory, max, unsatisfiable_classes);
+  }
+
+  /**
+   * Compute explanations for all most general unsatisfiable classes Note that this is a very naive
+   * implementation which assumes and acyclic class hierarchy.
+   *
+   * @param ontology the ontology to be tested
+   * @param reasoner the reasoner to be used to determine the unsatisfiable classes
+   * @param reasonerFactory the reasoner factory to be used to compute the explanations
+   * @param max maximum number of explanations to be computed
+   * @return a set of explanations
+   */
+  public static Set<Explanation<OWLAxiom>> explainMostGeneralUnsatisfiableClasses(
+      OWLOntology ontology, OWLReasoner reasoner, OWLReasonerFactory reasonerFactory, int max) {
+    List<OWLClass> unsatisfiable_classes =
+        new ArrayList<>(getMostGeneralUnsatisfiableClasses(reasoner, ontology));
+    return getUnsatExplanationsForClasses(ontology, reasonerFactory, max, unsatisfiable_classes);
+  }
+
+  private static Set<OWLClass> getMostGeneralUnsatisfiableClasses(
```

Comment:
Please move this to be with the other private methods (order public methods -> private methods)